e2e/analysis/analysis_download_dockerhub_test.go

```diff
@@ -0,0 +1,82 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package analysis_test
+
+import (
+	"fmt"
+	"os"
+	"runtime"
+
+	"github.com/ZupIT/horusec-devkit/pkg/utils/logger"
+	"github.com/ZupIT/horusec/e2e/analysis"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/ZupIT/horusec/internal/utils/testutil"
+)
+
+var _ = Describe("Run a complete horusec analysis when download tools from DockerHub", func() {
+	Describe("Running e2e tests when download from DockerHub", func() {
+		allTestsCases := analysis.NewTestCase(testutil.ExamplesPath)
+		isWindows := runtime.GOOS == "windows"
+		for idx := range allTestsCases {
+			testCase := allTestsCases[idx]
+			if isWindows && testCase.RequiredDocker {
+				logger.LogInfo("Tool ignored because is required docker: ", testCase.Tool, fmt.Sprintf("[%v/%v]", idx+1, len(allTestsCases)))
+				continue
+			}
+			logger.LogInfo("Preparing test for run e2e download from DockerHub: ", testCase.Tool, fmt.Sprintf("[%v/%v]", idx+1, len(allTestsCases)))
+			session, err := testCase.RunAnalysisTestCase()
+			if err != nil {
+				Fail(fmt.Sprintf("Error on run analysis Test Case when download from DockerHub: %v", err))
+			}
+			session.Wait(testutil.AverageTimeoutAnalyzeForExamplesFolder)
+			testCase.Command.Output = string(session.Out.Contents())
+			testCase.Command.ExitCode = session.ExitCode()
+
+			It(fmt.Sprintf("In download from DockerHub execute command without error for: %s", testCase.Tool), func() {
+				Expect(testCase.Command.ExitCode).Should(Equal(0))
+			})
+
+			It(fmt.Sprintf("In download from DockerHub validate is outputs expected exists on tool: %s", testCase.Tool), func() {
+				for _, outputExpected := range testCase.Expected.OutputsContains {
+					Expect(testCase.Command.Output).Should(
+						ContainSubstring(outputExpected),
+						fmt.Sprintf("The output [%s] not exist in output", outputExpected),
+						testCase.Command.Output,
+					)
+				}
+			})
+
+			It(fmt.Sprintf("In download from DockerHub validate is outputs not expected exists on tool: %s", testCase.Tool), func() {
+				for _, outputNotExpected := range testCase.Expected.OutputsNotContains {
+					Expect(testCase.Command.Output).ShouldNot(
+						ContainSubstring(outputNotExpected),
+						fmt.Sprintf("The output [%s] exist in output", outputNotExpected),
+						testCase.Command.Output,
+					)
+				}
+			})
+
+			if CurrentGinkgoTestDescription().Failed {
+				logger.LogInfo("Tool finished your analysis with status of failed: ", testCase.Tool, fmt.Sprintf("[%v/%v]", idx+1, len(allTestsCases)))
+				os.Exit(1)
+			}
+			logger.LogInfo("Tool finished your analysis with status of success: ", testCase.Tool, fmt.Sprintf("[%v/%v]", idx+1, len(allTestsCases)))
+		}
+	})
+
```

Comment:
We can reuse some logic from `analysis_build_locally_test.go`? Looks quite similar.

Reply:
i try but Ginkgo not allow use Go files for logic shared into tests, for use this will be necessary create new package inside e2e/analysis and call in tests. I beleave this is not good.

Reply:
Put on `testutil` package would be a option?

Reply:
but this logic is only usage in e2e tests, i'll think of a better solution

Reply:
i'm join two file into one, review again :)